cpp/modmesh/toggle/profile.hpp

```diff
@@ -156,6 +166,39 @@ class TimeRegistry
         return ostm.str();
     }
 
+    std::string detailed_report() const
+    {
+        std::ostringstream ostm;
+        /// Header
+        ostm
+            << "      " << total_call_count()
+            << " function calls in " << total_time()
+            << " seconds" << std::endl;
+        ostm << std::endl;
+        ostm
+            << std::setw(40) << "Function Name"
+            << std::setw(25) << "Call Count"
+            << std::setw(25) << "Total Time (s)"
+            << std::setw(25) << "Per Call (s)"
+            << std::setw(25) << "Cumulative Time (s)"
+            << std::setw(25) << "Per Call (s)"
+            << std::endl;
+
+        /// Body
+        for (auto it = m_entry.begin(); it != m_entry.end(); ++it)
+        {
+            ostm
+                << std::setw(40) << it->first
+                << std::setw(25) << it->second.count()
+                << std::setw(25) << it->second.time()
+                << std::setw(25) << it->second.time() / it->second.count()
+                << std::setw(25) << it->second.ctime()
+                << std::setw(25) << it->second.ctime() / it->second.count()
+                << std::endl;
+        }
+        return ostm.str();
+    }
```

Comment:
I add a new function to make the report clearly.
Additionally, by remaining the old feature `std::string report() const`, 
I think we could prevent from rewriting the pytest.